apps/icinga/src/main.cpp

```diff
@@ -0,0 +1,91 @@
+#include <Arduino.h>
+#include <otaserver.h>
+#include <kgfx.h>
+#include <HTTPClient.h>
+#include <ArduinoJSON.h>
+
+// CONFIGURATION
+// Please make sure you enabled and configured the new API in your Icinga 2 installation by either setting it up manually
+// or use the CLI command for the automatic setup, which also creates a root-user for the API with a random password (icinga2 api setup)
+// The default port of the Icinga 2 API is 5665, so the URL to go with would be like: https://myicingaserver.com:5665
+// Requires the following API permission(s): status/query
+// Read more about Icinga 2 API Setup: https://icinga.com/docs/icinga-2/latest/doc/12-icinga2-api/
+String icingaAPIURL = "https://hostname:5665";
+String icingaAPIUsername = "username";
+String icingaAPIPassword = "password";
+int refreshTimeInSeconds = 60;
+// -- CONFIGURATION
+
+OTAServer otaserver;
+KGFX ui;
+unsigned long lastTime = 0;
+TFT_eSprite textSpr = ui.createSprite(240, 240);
+HTTPClient http;
+JsonDocument json;
+
+void setup() {  
+  Serial.begin(460800);
+  Serial.println("Starting app");
+
+  otaserver.connectWiFi(); // DO NOT EDIT.
+  otaserver.run(); // DO NOT EDIT
+
```

Comment:
For configs you would add:  

```
  preferences.begin("app", true);
  icingaAPIURL = preferences.getString("icinga_url");
  icingaAPIUsername = preferences.getString("icinga_username");
  icingaAPIPassword = preferences.getString("icinga_password");
  preferences.end();
```

For development only, you can add the following code to populate the data in preferences:

``` 
  preferences.begin("app", true);
  // preferences.putString("icinga_url", "https://hostname:5665");
  // preferences.putString("icinga_username", "username");
  // preferences.putString("icinga_password", "password");
  icingaAPIURL = preferences.getString("icinga_url");
  icingaAPIUsername = preferences.getString("icinga_username");
  icingaAPIPassword = preferences.getString("icinga_password");
  preferences.end();
```

Then validate the values as you wish.

The [docs](https://developers.thekublet.com/docs/storage/nvs/) weren't clear on using this for dev/prod configs, will fix.